### PR TITLE
Update state tree correctly on status change

### PR DESCRIPTION
**Pull request by rehandalal**
r?

### CODE REVIEW

File: recipe-server/client/control/components/DisableRecipe.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ class DisableRecipe extends React.Component {
     event.preventDefault();
     dispatch(makeApiRequest('disableRecipe', { recipeId }))
     .then(() => {
-      dispatch(recipeDisabled(recipeId));
+      dispatch(singleRecipeReceived(recipeId));
```
Comment (andymikulski) [inline recipe-server/client/control/components/DisableRecipe.js]: `singleRecipeReceived` expects an object (the recipe)

File: recipe-server/client/control/components/EnableRecipe.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ class EnableRecipe extends React.Component {
     event.preventDefault();
     dispatch(makeApiRequest('enableRecipe', { recipeId }))
     .then(() => {
-      dispatch(recipeEnabled(recipeId));
+      dispatch(singleRecipeReceived(recipeId));
```
Comment (andymikulski) [inline recipe-server/client/control/components/EnableRecipe.js]: `singleRecipeReceived` expects an object (the recipe)

File: recipe-server/client/control/components/RecipeForm.js [hunk 3]
```diff
@@ -125,6 +129,8 @@ export class RecipeForm extends React.Component {
     const requestAuthorID = requestDetails && requestDetails.creator.id;
 
     const isUserViewingOutdated = !!(routeParams && routeParams.revisionId);
+    const isViewingLatestApproved = recipe && revision
+      && revision.revision_id === recipe.approved_revision_id;
```
Comment (Osmose) [inline recipe-server/client/control/components/RecipeForm.js]: Do we want a test for this value?

File: recipe-server/normandy/recipes/tests/test_api.py [hunk 4]
```diff
@@ -272,7 +272,7 @@ def test_it_can_enable_recipes(self, api_client):
         recipe = RecipeFactory(enabled=False, approver=UserFactory())
 
         res = api_client.post('/api/v1/recipe/%s/enable/' % recipe.id)
-        assert res.status_code == 204
+        assert res.status_code == 200
```
Comment (Osmose) [inline recipe-server/normandy/recipes/tests/test_api.py]: We should be testing for the content changes to this API as well as the status code change.

**Comment by rehandalal** [comment]
rebased to fix conflicts

**Pull request merged**